### ISSUE TITLE
Feature Request: Alternative paste command with sudo

### ISSUE THREAD

**Issue by og900aero** [TD-Sky/sudo.yazi#3]
Some users (example me) uses alternative paste command, because we love copy and move feature the following procedure:

Select files/folders with space, and
- press p for copy files/folders
- press P for move files/folders

The keymap.toml contains:

```
{ on = [ "p" ], run = [ "yank", "paste", "unyank" ], desc = "Paste command with copy" },
{ on = [ "P" ], run = [ "yank --cut", "paste", "unyank" ], desc = "Paste command with move" },

```
Could you accordingly create a paste_copy and paste_move argument?

**Comment by TD-Sky** [comment]
Are these just commands combination? I don't quite understand what you meant.

**Comment by og900aero** [comment]
> Are these just commands combination? I don't quite understand what you meant.

I would like to do the same with the sudo command. However, your solution unfortunately does not work that way, because after selection I have to indicate that you want to copy or move, and then the pasting. This is the default method, but I would like to reverse method works: select and paste with copy or paste with move.

**Comment by TD-Sky** [comment]
> because after selection I have to indicate that you want to copy or move

sudo.yazi pastes according to `cx.yanked.is_cut` which represent the yank status in the context.
Controlling the yank behavior via `yank [--cut]` should work.

I tried the keymaps you provided, it seems that you do fast rename files by them.
The reason I guess why sudo.yazi paste cannot work in the commands combination is there may be some internal mechanisms of yazi break yank-paste-unyank workflow.

**Comment by og900aero** [comment]
> > because after selection I have to indicate that you want to copy or move
> 
> sudo.yazi pastes according to `cx.yanked.is_cut` which represent the yank status in the context. Controlling the yank behavior via `yank [--cut]` should work.
> 
> I tried the keymaps you provided, it seems that you do fast rename files by them. The reason I guess why sudo.yazi paste cannot work in the commands combination is there may be some internal mechanisms of yazi break yank-paste-unyank workflow.

Yes. The mechanism I outlined would probably require a paste_copy and paste_cut function: Maybe:

```
local function sudo_paste_copy(value)
    local args = sudo_cmd()
    extend_list(args, { "cp", "-rf" })
    extend_iter(args, list_map(value.yanked, ya.quote))
    extend_list(args, { "-t", "./" })
    execute(args)
end
local function sudo_paste_move(value)
    local args = sudo_cmd()
    extend_list(args, { "mv", "-f" })
    extend_iter(args, list_map(value.yanked, ya.quote))
    extend_list(args, { "-t", "./" })
    execute(args)
end
```
But I don't know how to rewrite this so I can reference it.

```
local get_state = ya.sync(function(_, cmd)
    if cmd == "paste" or cmd == "link" then
        local yanked = {}
        for _, url in pairs(cx.yanked) do
            table.insert(yanked, tostring(url))
        end

        if #yanked == 0 then
            return {}
        end

        return {
            kind = cmd,
            value = {
                is_cut = cx.yanked.is_cut,
                yanked = yanked,
            },
        }
```

**Comment by TD-Sky** [comment]
I have tried writing a pure lua script and calling it with `sudo` for your demand, which introduced more trouble because I cannot use luarocks.

However, I don't wanna maintain a bash shell script at all.

Can I use nushell in sudo.yazi? Nushell scripts is more readable and have higher maintainability. The only thing you need to do is installing nu.

**Comment by og900aero** [comment]
> I have tried writing a pure lua script and calling it with `sudo` for your demand, which introduced more trouble because I cannot use luarocks.
> 
> However, I don't wanna maintain a bash shell script at all.
> 
> Can I use nushell in sudo.yazi? Nushell scripts is more readable and have higher maintainability. The only thing you need to do is installing nu.

Yes u can of course.

**Comment by TD-Sky** [comment]
I implement all your requests in #4. Please check it (especially the README.md) and report the bugs your find.

**Comment by og900aero** [comment]
> I implement all your requests in #4. Please check it (especially the README.md) and report the bugs your find.

I checked the README.md, but I dont know how works. There is no distinction between the copy+paste and move+paste commands. There is also 1 type of paste command.

**Comment by TD-Sky** [comment]
> > I implement all your requests in #4. Please check it (especially the README.md) and report the bugs your find.
> 
> I checked the README.md, but I dont know how works. There is no distinction between the copy+paste and move+paste commands. There is also 1 type of paste command.

Just copy/cut and then paste. We misjudged mechanism of yazi before.

**Comment by og900aero** [comment]
> > > I implement all your requests in #4. Please check it (especially the README.md) and report the bugs your find.
> > 
> > 
> > I checked the README.md, but I dont know how works. There is no distinction between the copy+paste and move+paste commands. There is also 1 type of paste command.
> 
> Just copy/cut and then paste. We misjudged mechanism of yazi before.

But with nu shell it works exactly as it did before. I have to select it for copying or moving, and then paste with sudo. But the original (my) usage is exactly the opposite of this, that I just select the file, and there are two methodes of the pasting (copy+paste and move+paste), see the first comment.

**Comment by TD-Sky** [comment]
> > > > I implement all your requests in #4. Please check it (especially the README.md) and report the bugs your find.
> > > 
> > > 
> > > I checked the README.md, but I dont know how works. There is no distinction between the copy+paste and move+paste commands. There is also 1 type of paste command.
> > 
> > 
> > Just copy/cut and then paste. We misjudged mechanism of yazi before.
> 
> But with nu shell it works exactly as it did before. I have to select it for copying or moving, and then paste with sudo. But the original (my) usage is exactly the opposite of this, that I just select the file, and there are two methodes of the pasting (copy+paste and move+paste), see the first comment.

Sorry, I forget the original request ...... But now `sudo paste` can choose a usable destination name.

Well, I will open a new PR to resolve this.

**Comment by TD-Sky** [comment]
Install the latest sudo.yazi (via git) and try the config below:

```
// Don't add yank
{ on = "p", run = ["escape --visual", "yank", "plugin sudo --args='paste'"], desc = "Paste command with copy" },
{ on = "P", run = ["escape --visual", "yank --cut", "plugin sudo --args='paste'"], desc = "Paste command with move" },
```

The reason why it works: https://github.com/sxyazi/yazi/issues/1553

**Comment by og900aero** [comment]
> Install the latest sudo.yazi (via git) and try the config below:
> 
> ```
> // Don't add yank
> { on = "p", run = ["escape --visual", "yank", "plugin sudo --args='paste'"], desc = "Paste command with copy" },
> { on = "P", run = ["escape --visual", "yank --cut", "plugin sudo --args='paste'"], desc = "Paste command with move" },
> ```
> 
> The reason why it works: [sxyazi/yazi#1553](https://github.com/sxyazi/yazi/issues/1553)

Whoaaa! It is working great! Thank u :)
